cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectAssemblyOperatorMockTest.java

```diff
@@ -156,4 +166,51 @@ public void testReconcileCreateAndUpdate(VertxTestContext context) {
 
     }
 
+    @Test
+    public void testPauseReconcile(VertxTestContext context) {
+        setConnectResource(new KafkaConnectBuilder()
+                .withMetadata(new ObjectMetaBuilder()
+                        .withName(CLUSTER_NAME)
+                        .withNamespace(NAMESPACE)
+                        .withLabels(TestUtils.map("foo", "bar"))
+                        .withAnnotations(singletonMap("strimzi.io/pause-reconciliation", "true"))
+                        .build())
+                .withNewSpec()
+                .withReplicas(replicas)
+                .endSpec()
+                .build());
+        KafkaConnectApi mock = mock(KafkaConnectApi.class);
+        when(mock.list(anyString(), anyInt())).thenReturn(Future.succeededFuture(emptyList()));
+        when(mock.listConnectorPlugins(anyString(), anyInt())).thenReturn(Future.succeededFuture(emptyList()));
+
+        Checkpoint async = context.checkpoint();
+        createConnectCluster(context, mock, true)
+                .onComplete(context.succeeding())
+                .compose(v -> {
+                    LOGGER.info("Reconciling again -> update");
+                    return kco.reconcile(new Reconciliation("test-trigger", KafkaConnect.RESOURCE_KIND, NAMESPACE, CLUSTER_NAME));
+                })
+                .onComplete(context.succeeding(v -> context.verify(() -> {
+                    Resource<KafkaConnect> resource = Crds.kafkaConnectOperation(mockClient).inNamespace(NAMESPACE).withName(CLUSTER_NAME);
+                    io.strimzi.operator.cluster.TestUtils.waitForStatus(resource, CLUSTER_NAME, s -> {
+                        if (s.getStatus() == null) {
+                            return false;
+                        }
+                        List<Condition> conditions = s.getStatus().getConditions();
+                        boolean conditionFound = false;
+                        if (conditions != null && !conditions.isEmpty()) {
+                            for (Condition condition: conditions) {
+                                if ("ReconciliationPaused".equals(condition.getType())) {
+                                    conditionFound = true;
+                                    break;
+                                }
+                            }
+                        }
+                        return conditionFound;
+                    });
```

Comment:
Isn't this too complicated? You know the previous reconciliation ended so you know the status should be set, or? So why the need to use `waitForStatus`? Is it because of some race-conditions in MockKube?